skimage/future/detect_obj/cascade.pyx

```diff
@@ -0,0 +1,832 @@
+        """Get the valid scale multipliers for the original window size.
+
+        The function takes the minimal size of window and maximum size of
+        window as interval and finds all the multipliers that will give the
+        windows which sizes will be not less than the min_size and not bigger
+        than the max_size.
+
+        Parameters
+        ----------
+        min_size : typle (int, int)
+            Minimum size of window for which to search the scale factor.
+        max_size : typle (int, int)
+            Maximum size of window for which to search the scale factor.
+        scale_step : float
+            The scale by which the search window is multiplied
+            on each iteration.
+
+        Returns
+        -------
+        scale_factors : 1-D floats ndarray
+            The scale factors that give the window sizes that are in the
+            specified interval after multiplying the search window.
+        """
+
+        current_size = np.array((self.window_height, self.window_width))
+        min_size = np.array(min_size, dtype=np.float32)
+        max_size = np.array(max_size, dtype=np.float32)
+
+        row_power_max = math.log(max_size[0]/current_size[0], scale_step)
+        col_power_max = math.log(max_size[1]/current_size[1], scale_step)
+
+        row_power_min = math.log(min_size[0]/current_size[0], scale_step)
+        col_power_min = math.log(min_size[1]/current_size[1], scale_step)
+
+        mn = max(row_power_min, col_power_min, 0)
+        mx = min(row_power_max, col_power_max)
+
+        powers = np.arange(mn, mx)
+
+        scale_factors = np.power(scale_step, powers, dtype=np.float32)
+
+        return scale_factors
+
+    def _get_contiguous_integral_image(self, img):
+        """Get a c-contiguous array that represents the integral image.
+
+        The function converts the input image into the integral image in
+        a format that is suitable for work of internal functions of
+        the cascade classifier class. The function converts the image
+        to gray-scale float representation, computes the integral image
+        and makes it c-contiguous.
+
+        Parameters
+        ----------
+        img : 2-D or 3-D ndarray
+            Ndarray that represents the input image.
+
+        Returns
+        -------
+        int_img : 2-D floats ndarray
+            C-contiguous integral image of the input image.
+        """
+        if len(img.shape) > 2:
+            img = rgb2gray(img)
+        int_img = integral_image(img)
+        int_img = np.ascontiguousarray(int_img, dtype=np.float32)
+
+        return int_img
+
+
+    def detect_multi_scale(self, img, float scale_factor, float step_ratio,
+                           min_size, max_size, min_neighbour_amount=4,
+                           intersection_score_threshold=0.5):
+        """Search for the object on multiple scales of input image.
+
+        The function takes the input image, the scale factor by which the
```

Comment:
You can remove this paragraph (duplicates the parameter description below), and rather expand on what the function does.